src/client/components/Tooltip/Tooltip.tsx

```diff
@@ -87,18 +88,12 @@ export const Tooltip: React.FC<TooltipProps> = ({ body }) => {
   const ref = useRef<HTMLDivElement>(null)
   const [isVisible, setVisible] = useState(false)
 
-  useEffect(() => {
-    if (!isVisible) return
-
-    const handleClickOutside = (event: TouchEvent) => {
-      if (ref.current && !ref.current.contains(event.target as Node)) {
-        setVisible(false)
-      }
+  const handleClickOutside = useCallback(() => {
+    if (isVisible) {
+      setVisible(false)
```

Comment:
Can't this be simplified to:

```
const handleClickOutside = useCallback(() => setVisible(false), [])
```

- `setVisible` doesn't need to be included in the dependency array
- after clicking outside, `isVisible` should always be `false`

Reply:
Yeah, makes sense. Thanks

Reply:
🤔 I believe I can remove `setVisible` from dependency array but `isVisible` I can't. If we look back to the [previous](https://github.com/HedvigInsurance/web-onboarding/blob/master/src/client/components/Tooltip/Tooltip.tsx#L90) implementation, we're returning early in case it's not visible. I believe that's happening due animation. We're not removing `TooltipContainer` from the dom so we can animate it's `visibility`. However since it's being unmounted `handleClickOutside` would still be attached and if we, while using mobile, click on tooltip icon `handleClickOutside` will be executed and, as we would be setting `isVisible` always to false, the tooltip content would never be shown.

Reply:
Actually, it turn out `useClickOutside` does not apply to `Tooltip` so I'm reverting it to previous implemeation.